initial notes on starting the swtpm (#23)
distributed-technologies/wiki: TPM/swtpm.md

> 
> 
> # SWTPM

Reviewer: ```suggestion
# SWTPM
```

> The following explains how to launch and use the `swtpm`. 

Reviewer: ```suggestion
The following explains how to launch and use `swtpm`. 
```

> The standard software implementation of a TPM is [swtpm](https://github.com/stefanberger/swtpm). A recipe for compiling it included in the meta-security/meta-tpm layer of the open-embedded project. 

Reviewer 2: ```suggestion
The standard software implementation of a TPM is [swtpm](https://github.com/stefanberger/swtpm). A recipe for compiling it is included in the meta-security/meta-tpm layer of the open-embedded project. 
```

> The standard software implementation of a TPM is [swtpm](https://github.com/stefanberger/swtpm). A recipe for compiling it included in the meta-security/meta-tpm layer of the open-embedded project. 

Reviewer: ```suggestion
The standard software implementation of a TPM is [swtpm](https://github.com/stefanberger/swtpm). A recipe for compiling it is included in the [meta-security/meta-tpm layer](https://git.yoctoproject.org/cgit/cgit.cgi/meta-security/tree/meta-tpm/recipes-tpm/swtpm) of the OpenEmbedded project. 
```
Let's add a link to the recipe.

For testing the features which rely on a `TPM` we can compile it as a native tool on the host and start it before launching the image with `runqemu`. I.e. we compile it with:
```
bitbake swtpm-native
```

> NOTE: We are currently in the process of updating the version of swtpm: 0.5.2 --> 0.6.0. This will fix some issues with dependencies on python.

Reviewer: We should link to the upstream patch here.

Author: There is no patch yet. I am trying to work with the auto-upgrade-helper as recommended [here](https://www.yoctoproject.org/docs/latest/mega-manual/mega-manual.html#gs-upgrading-recipes). But it is not as easy as it looks.

> After the swtpm is compiled the binaries are located in `${WORKDIR}/swtpm-native`.

Reviewer: ```suggestion
After swtpm is compiled the binaries are located in `${WORKDIR}/swtpm-native`.
```

> The [QEMU documentation](https://qemu-project.gitlab.io/qemu/specs/tpm.html#the-qemu-tpm-emulator-device) suggests starting the `swtpm` with 

Reviewer: ```suggestion
The [QEMU documentation](https://qemu-project.gitlab.io/qemu/specs/tpm.html#the-qemu-tpm-emulator-device) suggests starting `swtpm` with 
```
